shop/src/main/java/ro/msg/learning/shop/controller/LocationController.java

```diff
@@ -0,0 +1,44 @@
+package ro.msg.learning.shop.controller;
+
+import lombok.AllArgsConstructor;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.validation.annotation.Validated;
+import org.springframework.web.bind.annotation.*;
+import ro.msg.learning.shop.dto.LocationDto;
+import ro.msg.learning.shop.exception.LocationNotFoundException;
+import ro.msg.learning.shop.service.LocationService;
+
+import java.util.List;
+import java.util.UUID;
+
+@RequestMapping("/locations")
+@Validated
+@RestController
+@AllArgsConstructor
+public class LocationController {
+
+    private final LocationService locationService;
+
+
+
+    @GetMapping
+    public ResponseEntity<List<LocationDto>> getAll() {
+        return new ResponseEntity<List<LocationDto>>(locationService.findAllLocations(), HttpStatus.OK);
+    }
+
+    @DeleteMapping("/{id}")
+    public ResponseEntity<String> delete(@PathVariable("id") UUID locationId) throws LocationNotFoundException {
```

Comment:
It is not ok to throw exceptions and not handle them.
So you throw an exception in the service part which is very good but then we have 2 options to deal with them:
1. Try catch blocks
2. Controller advice: we define for every exception what happens if that is thrown.
Reference: https://www.baeldung.com/exception-handling-for-rest-with-spring#controlleradvice